packages/slate-dev-environment/src/index.js

```diff
@@ -91,6 +91,43 @@ if (isBrowser) {
   }
 }
 
+/**
+ * Array of regular expression matchers and their API version
+ *
+ * @type {Array}
+ */
+
+const ANDROID_API_VERSIONS = [
+  [/^9([.]0|)/, 28],
+  [/^8[.]1/, 27],
+  [/^8([.]0|)/, 26],
+  [/^7[.]1/, 25],
+  [/^7([.]0|)/, 24],
+  [/^6([.]0|)/, 23],
+  [/^5[.]1/, 22],
+  [/^5([.]0|)/, 21],
+  [/^4[.]4/, 20],
+]
+
+/**
+ * get the Android API version from the userAgent
+ *
+ * @return {number} version
+ */
+
+function getAndroidApiVersion() {
+  if (os !== 'android') return null
+  const { userAgent } = window.navigator
+  const matchData = userAgent.match(/Android\s([0-9\.]+)/)
+  if (matchData == null) return null
```

Comment:
How about, if the useragent matches Android but not the table, just assume the highest API? That way it's somewhat compatible with newer versions without sites having to upgrade their code. Older versions than Android 4 are probably not that likely to visit JS heavy sites.

Reply:
This is probably a good idea, but there is a trade off at play. As implemented, if there's a new android major that comes out, the plugin won't detect that it should apply, which will most certainly break functionality. 

If we default to the highest version as suggested, we have at least some chance of hiding incompatibility problems, but there will probably be some insidious issues in play. 

Partially working is probably better than not working at all.

Reply:
I agree with the concerns. That said, I recommend we keep it this way for now.

A major version number update for Android has previously always broken Slate so we don't gain much from future proofing major version numbers.

I'm tempted to lean the other way and break Android loudly for undetected version numbers by throwing an error that this version of Android isn't supported. Although I think that's not exactly straightforward  because we don't want to disable Android in the case where somebody is using Slate on Android only to view content. We don't want to throw errors in that case because displaying content through Slate on Android will work fine, it's just the editing that breaks.

Let me know if you think it's worth holding off a merge at this point. Otherwise, I'll merge in a few days.

Reply:
Forgot to say thank you for doing a code review on this PR. Very much appreciated.

Reply:
I'm personally comfortable with merging as is, with this review we at least have some historical documentation about the decision to loudly break if there's a major android version increment.

I know ian's busy but did we get his opinion on that?

Reply:
I’ll send him a message. He is the one that suggested moving this code into slate-dev-environment as a comment in the Android PR. He didn’t specifically approve of putting into a separate PR.

Reply:
Actually, I'm thinking of the many websites that will be semi abandoned in a few years and they will all have broken Slates, whereas maybe the future android version might have real input level 2 support.

I'd rather warn on newer versions, saying something like "Slate was not yet tested with this Android version, it might not work". You can just parse the version and see if it's higher than the current highest…

Reply:
@wmertens I'll update this PR to add INPUT_EVENTS_LEVEL_2 for non-matching Android API versions. I'm not going to do the Android warning here right now since this isn't the PR for Android support. That's a separate PR.

Reply:
Added INPUT_EVENTS_LEVEL_2 support for non-matching Android.

Reply:
@CameronAckermanSEL I've reached out to @ianstormtaylor about this PR but haven't gotten a reply yet. I'm assuming he's busy.

I asked him some questions with respect to contributing as I've noticed that I'm the only one merging PRs other than him and whether he wanted me to continue with that as well as offering to write up some documentation on rules for accepting PRs.

Since he hasn't replied yet and hasn't asked me to stop in the past despite having rejected two of my PRs (he's allowed the rest of them) I'm going to assume it's okay. He also has the option to revert them and he controls publishing to NPM so none of these updates will get into the wild without his publishing them explicitly. I'm also aware of why he rejected the last two PRs so I'm making sure that he doesn't have to say the same thing twice.

I'll let this PR sit for another day or so with the revision. If anybody sees anything wrong with the update, please let me know. I haven't tested this code against Android as I don't have Android Studio setup at my home computer (where I'm working today) but the change is pretty straightforward.